Source/Dafny/Resolver.cs

```diff
@@ -16942,6 +16948,10 @@ class MethodCallInformation {
           if (formals.Count != 1) {
             what += " " + j;
           }
+          if (formal.HasName && !(formal is ImplicitFormal)) {
```

Comment:
Similarly, here it seems strange to include both the index and name.

Btw, what's a situation in which there's no name? And what's the implicitFormal situation?

Reply:
Sure, here are the situations I encountered in the CI.

```
datatype List = Nil | Cons(nat, List)
```
Here the formal associated to "nat" does not have any name, it makes no sense to display the generated name "_p0".

A second example:
```
function f(callback: nat -> nat): nat {
  callback(-1)
}
```
Here a formal is specially created for the application of callback, because call sites need formals. This formal is implicit and has the name "_#p0".

Last example:
```
least lemma G[ORDINAL](x: int)
    ensures x == 8
  {
    H(x);  // error: cannot call from [ORDINAL] to [nat]
  }

  least lemma H[nat](x: int)
    ensures x == 8
  {
    G(x);  // error: cannot call from [nat] to [ORDINAL]
  }
```

Here least lemmas are implicitly provided another first formal, thus an implicit formal, which is named `_k` (see [here](https://dafny-lang.github.io/dafny/DafnyRef/DafnyRef#sec-friendliness) for an explanation of why these variables are useful) which makes little sense to report since the name is not part of the AST.

Reply:
Thanks